### PR TITLE
added check for docker or kubernetes platform

### LINKED ISSUE #3444: [mesheryctl] Check for invalid platforms on system start
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior
Currently specifying an invalid platform using the `-p` flag throws an error that the platform is an invalid after updating the platform in the `meshconfig`.  

#### Desired Behavior
If a platform other than docker or kubernetes is specified, it should throw an error message and not update the `meshconfig`

#### Screenshots/Logs
image

#### Environment

- Host OS: Mac Linux Windows

#### Implementation
Adding a check in `start.go` when updating the platform.  

---

#### Contributor Resources

- [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by nehaLohia27**
Signed-off-by: Neha Lohia <nehapithadiya444@gmail.com>

**Description**

This PR fixes #3444 

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Review by piyushsingariya — DISMISSED** [review]
Looks Good! Nice work @nehaLohia27

File: mesheryctl/internal/cli/root/system/start.go [hunk 1]
```diff
@@ -102,8 +102,15 @@ func start() error {
...
-		currCtx.SetPlatform(utils.PlatformFlag)
+	if utils.PlatformFlag == "docker" || utils.PlatformFlag == "kubernetes" {
+		currCtx.Platform = utils.PlatformFlag
+		err := utils.ChangePlatform(mctlCfg.CurrentContext, currCtx)
+
+		if err != nil {
+			return err
+		}
+	} else {
+		return errors.New(fmt.Sprintf("the platform %s is not supported currently. The supported platforms are:\ndocker\nkubernetes\nPlease check %s/config.yaml file.", utils.PlatformFlag, utils.MesheryFolder))
```
Comment (metonymic-smokey) [inline mesheryctl/internal/cli/root/system/start.go:113]: Looks good for the most past @nehaLohia27 !

@piyushsingariya should the location of the config file be hardcoded to the default? I think the message should display the location of the config file when the user changes it using the flag too...
What do you think?

Comment (piyushsingariya) [reply mesheryctl/internal/cli/root/system/start.go:113]: right! @nehaLohia27 can you implement this?

Comment (nehaLohia27) [reply mesheryctl/internal/cli/root/system/start.go]: @piyushsingariya @metonymic-smokey i have implemented the code. But PR keeps failing. not sure what the problem. can you please help ?

**Review by leecalcote — APPROVED** [review]

File: mesheryctl/internal/cli/root/system/start.go (hunk 1 above)
Comment (leecalcote) [reply mesheryctl/internal/cli/root/system/start.go]: Great catch, @metonymic-smokey

**Comment by leecalcote** [comment]
@nehaLohia27 I tweaked the formatting and language used.

**Review by leecalcote — APPROVED** [review]
Good work, @nehaLohia27. Thanks for seeing this through!

**Comment by nehaLohia27** [comment]
thanks @leecalcote @hexxdump @metonymic-smokey @piyushsingariya  for the help.

**Pull request merged**